### PR TITLE
Some design changes

### PR DESCRIPTION
**Pull request by jbtbnl**
See issue https://github.com/owncloud/contacts/issues/289

### CODE REVIEW

**Comment by jancborchardt** [comment]
@jbtbnl wohoo, really good changes! From the first look I have 2 remarks:
- let’s not use a controls bar just for the 2 buttons. We want to get rid of controls bars – it’s only not possible in Files because of the breadcrumbs. Simply put the buttons topmost in the left sidebar (app-navigation)
- contacts seems to use a lot of custom stuff for app-navigation and such. Lots of that is already defined in [core/apps.css](https://github.com/owncloud/core/blob/master/core/css/apps.css) – let’s not duplicate stuff but try to take these styles as much as possible instead.

Check it out too @owncloud/designers @tanghus

File: css/contacts.css [hunk 1]
```diff
@@ -85,9 +91,10 @@
 #app-navigation img.shared:hover { opacity: 1; }*/
 #app-navigation input[type="text"] {
 	display: block;
-	width: 200px;
+	width: 200px !important;
```
Comment (Niduroki) [inline css/contacts.css]: Try to avoid `!important` if possible.

File: css/contacts.css [hunk 2]
```diff
@@ -125,6 +138,25 @@
 #grouplist > li {
 	height: 44px;
 }
+#grouplist .open {
+	margin-top: 5px !important;
```
Comment (Niduroki) [inline css/contacts.css]: Try to avoid `!important` if possible.

File: css/contacts.css [hunk 3]
```diff
@@ -125,6 +138,25 @@
...
+}
+
+.hidden-on-load {
+	display: none;
+}
+
+/* Controls */
+.new {
+	padding: 0px !important;
```
Comment (Niduroki) [inline css/contacts.css]: Just `0` is enough.
Also try to avoid `!important` if possible.

File: css/contacts.css [hunk 4]
```diff
@@ -125,6 +138,25 @@
...
+}
+.new a {
+	padding: 14px 10px;
+	line-height: 34px;
+}
+.actions {
+	padding-left: 4px !important;
```
Comment (Niduroki) [inline css/contacts.css]: Try to avoid `!important` if possible.

File: css/contacts.css [hunk 5]
```diff
@@ -636,19 +671,24 @@ fieldset.adr ul li input.label { margin-top: -4px !important; }
 }
 #contactsHeader .action {
 	margin: 0;
-	padding: 0 5px;
+	padding: 0 0px;
```
Comment (Niduroki) [inline css/contacts.css]: `0 0`, maybe even `0` should be enough here.

File: css/contacts.css [hunk 6]
```diff
@@ -670,18 +710,16 @@ tbody tr.contact.active, tbody tr.contact:hover {
...
-#contactlist tr > td:not(.tel):not(.adr):not(.actions) {
-	max-width: 270px;
-	min-width: 200px;
+#contactlist tr > td.adr {
+	min-width: 80px;
 }
-#contactlist tr > td.tel {
-	max-width: 180px;
-	min-width: 180px;
+#contactlist tr > td.adr {
```
Comment (Niduroki) [inline css/contacts.css]: Duplicate of 717&718.

File: css/jquery.ocaddnew.css [hunk 7]
```diff
@@ -20,85 +20,26 @@
...
-	font-weight: bold;
-	font-size: 180%;
-	padding-right: 5px;
-}
-
-.oc-addnew input[type=text] {
-	width: 120px !important;
-	margin-top: 15px;
-	height: 30px !important;
+	background-image: none !important;
```
Comment (Niduroki) [inline css/jquery.ocaddnew.css]: Try to avoid `!important` if possible.

File: css/jquery.ocaddnew.css [hunk 8]
```diff
@@ -20,85 +20,26 @@
...
 }
 
-.oc-addnew select {
-	width: 133px;
-	margin-top: 1px !important;
-	height: 30px;
-	background-color: #eee;
+.oc-addnew li[display=list-item] {
```
Comment (Niduroki) [inline css/jquery.ocaddnew.css]: empty?

File: css/jquery.ocaddnew.css [hunk 9]
```diff
@@ -20,85 +20,26 @@
...
-	display: inline-block;
-	float: left;
-	border-top-right-radius: 0;
-	border-bottom-right-radius: 0;
-	border-right: 0;
+.oc-addnew input[type=text] {
+	width: 200px !important;
+	margin: 0px !important;
+	border-width: 0px;
+	height: 34px !important;
```
Comment (Niduroki) [inline css/jquery.ocaddnew.css]: Try to avoid `!important` if possible.

File: css/jquery.ocaddnew.css [hunk 10]
```diff
@@ -20,85 +20,26 @@
...
-	border-bottom-right-radius: 5px; 
-}
-
-.oc-addnew .create-button {
-	border-radius: 0;
-	background-image: url('%appswebroot%/contacts/img/checkmark-gray.svg');
-	border-top-right-radius: 5px; 
-	border-bottom-right-radius: 5px; 
+	height: 34px !important;
+	margin: 0px !important;
```
Comment (Niduroki) [inline css/jquery.ocaddnew.css]: Just `0` should be enough here.
Also try to avoid `!important` if possible.

File: js/contacts.js [hunk 11]
```diff
@@ -2281,7 +2281,7 @@ OC.Contacts = OC.Contacts || {};
 			revert: 'invalid',
 			//containment: '#content',
 			helper: function (e,ui) {
-				return $(this).clone().appendTo('body').css('zIndex', 5).show();
+				return true;
```
Comment (Niduroki) [inline js/contacts.js]: This doesn't seem helpful.

**Comment by jbtbnl** [comment]
@jancborchardt there are some more then two buttons in the controls bar, I'm not sure where to keep them if the controls bar is discouraged.

untitled

@Kondou-ger I will take a look at your comments, some !important declarations might be obsolete but some certainly are.

**Comment by jancborchardt** [comment]
@jbtbnl all these multiselect actions should be in the header bar of the list – just like in the Files app when you select multiple files. Essentially the table header gets overlayed, you know what I mean?

Btw while you’re at it, can you remove these custom checkbox styles? They look really really strange. :)

Big thanks for your design contributions, they are much appreciated especially as I had little time to check out the Contacts app recently. Don’t hesitate to ask me if you have any questions. We also have http://owncloud.org/developer/design/ if you want to dive into ownCloud core design. :)

**Comment by tanghus** [comment]
I don't see the need for adding an extra top bar, or for removing the Add buttons from the side bar. As for the latter it will diverge from the style used in News, Notes etc.
The changes to the 'lower' top bar looks great though @jbtbnl

**Comment by jancborchardt** [comment]
@tanghus exactly what I said. ;) Thanks for seconding.

**Comment by jbtbnl** [comment]
Proposed design for contact editing:

untitled

**Comment by jancborchardt** [comment]
@jbtbnl nice proposal! Some small things I would change:
- the name field should be bigger. Similar to the event detail view in the calendar, for the event title
- the labels »email« »address« etc should not be bold. The focus should be on the field contents always.
- the labels and inputs being below each other break the flow of reading the info. The labels should always be to the left of the inputs. The additional info like work or home could be greyed out to the right of the field.
- the »Add field« dropdown needs to be flush with the other fields.
- close should be on the top right
- delete should be on the bottom right

What do you think?

**Comment by jancborchardt** [comment]
Btw @stefan-niedermann check this out since you also did a mockup in https://github.com/owncloud/contacts/issues/93#issuecomment-30432777

@jbtbnl @stefan-niedermann we could do an IRC meeting or voice call if you’re interested in contributing more to ownCloud design. :) We need to get coordinated in that case.

**Comment by tanghus** [comment]
@jbtbnl Looks good!

Please have in mind:
- Property blocks (email, phone etc) must fill horizontally depending on viewport width.
- Address property is multi valued https://tools.ietf.org/html/rfc2426#section-3.2.1 and therefore now expands when clicked. This should maybe be redesigned to do smth like the name field, where you have a one-line input for [LABEL](https://tools.ietf.org/html/rfc2426#section-3.2.2) - which was just recently requested in #284 - and an option to expand to show the different address fields. Some simple logic could - when requested by user - sync between LABEL and ADR. Not as complex as described in #284 unless someone volunteers ;)
- Name is also multi-valued: https://tools.ietf.org/html/rfc2426#section-3.1.1 and https://tools.ietf.org/html/rfc2426#section-3.1.2

> the »Add field« dropdown needs to be flush with the other fields.

@jancborchardt I'm not sure what you mean by that?

**Comment by jancborchardt** [comment]
@tanghus: sorry, I meant »aligned«. The »Add field« dropdown should be aligned to the left with the value inputs above and have the same width.

**Comment by jbtbnl** [comment]
In response to @jancborchardt:

**the name field should be bigger. Similar to the event detail view in the calendar, for the event title**
Changed it :)

**the labels »email« »address« etc should not be bold. The focus should be on the field contents always.**
Good point, done.

**the labels and inputs being below each other break the flow of reading the info. The labels should always be to the left of the inputs. The additional info like work or home could be greyed out to the right of the field.**
There is a lot of discussion about this in general and quite some research has been done on label placement. Most experiments conclude by the use of eye-tracking that labels on top of input fields are easier to process.

**the »Add field« dropdown needs to be flush with the other fields.**
Yes, probably I'll make the other three buttons slightly larger because of the extra height needed by the dropdown.

**close should be on the top right, delete should be on the bottom right**
Good idea, and then not a button but "Close x" and "Delete x" as in Files?

In response to @tanghus:

**Property blocks (email, phone etc) must fill horizontally depending on viewport width.**
Should we have a max-width for this? I'm on a 1920 x 1200px display, so filling horizontally all the way will look a bit silly :)
Is there a minimum and maximum supported resolution defined for owncloud?

**Address & name multi-valued**
Currently the name and address fields expand as they used to:
untitled untitled

**Comment by tanghus** [comment]
> Should we have a max-width for this?

Yes, that's probably a good idea. I played around a bit with media queries in the css, but I'll have to admit that UI/layout stuff is more than my nervous system can take, so I'm **very** happy to get some new eyes on that.

> Currently the name and address fields expand as they used to:

Nice!

**Comment by jbtbnl** [comment]
@tanghus @jancborchardt please take a look at the current state of this branch.
untitled

**Comment by tanghus** [comment]
@jbtbnl Thanks, gonna test later tonight. One thing: In the screen shot the add buttons are gone. I suppose because the jquery.ocaddnew.css is almost empty?

**Comment by tanghus** [comment]
The design looks **very** good! The are a few regressions though (some of them may be Firefox specific):
- It seems contacts that are not in any group doesn't get their pic shown in the list if they have one, and if they haven't none is generated for them. I can't quite fathom why that should make a difference, but it's easily reproducible.
- It seems the picture loading for open contacts is broken? I see no pictures for any contacts.
- Merge button shows even if only one contact is selected.
- The sort and groups dropdowns in the header are some pixels too short to show content.
- When one or more contacts are selected, the header shifts to double height.

None selected:
list-non-selected
One selected:
list-one-selected
- The add button for address books has the text shifted to the left:
  addressbook-add-pixel-off
- When opening a contact with a lot of properties, they are all listed vertically wasting a lot of screen real estate and causing a lot of scrolling. The example below has just as much off screen that can't be seen:
  open_contact_white_space

Don't let these minor issues hold you back; I really like what you have done so far :)

**Comment by jancborchardt** [comment]
@jbtbnl niiice! :) Few points about the screenshot:
- Instead of saying »Display name« in the header, can you put there (like it is in the Files app) »Jan, Thomas, Frank and 2 more«, or however many are selected? And show the first names of the first two or three just as reminder. 
- The »Groups« multiselect action has the Share icon as icon. And the »Merge« action has the groups icon. We need a better merge icon it seems. :) Maybe use the »Edit« pencil?

**Comment by jbtbnl** [comment]
@tanghus thanks for your feedback, I will do some cross-browser testing!
The pictures of not grouped contacts not showing is a bug in master, not a regression :) In master it's only visible with contacts that have a photo.

@jancborchardt do you know the source of the icons? I guess we need an icon originating from the same design set.

**Comment by jbtbnl** [comment]
@tanghus **When opening a contact with a lot of properties, they are all listed vertically wasting a lot of screen real estate and causing a lot of scrolling. The example below has just as much off screen that can't be seen:**

One of the causes is the "preferred" checkbox, which is shown on top of "type" selector, causing extra whitespace after each email input. What OS are you using? I can't reproduce this on Windows 7 with Firefox. Please tell me the width in pixels of the checkbox on your OS.

**Comment by jancborchardt** [comment]
@jbtbnl I made the icons. ;) As I said, best use the edit/pencil icon for merge for now. It fits the function.

**Comment by jancborchardt** [comment]
@jbtbnl just started using this branch on my production installation – soooo much better! Thank you! :)

And a few more remarks rolling in:
- [x] the number of contacts in a group is in () parentheses. No idea why, and these can be removed.
- [x] the input on the bottom right of the user image is cool. But the whole user image should be clickable to select the contact. You can solve this like it’s done in Files: With a label which has enough size/padding to cover that area.
- [ ] the inputs and labels should really not be in the same column – this creates a mix of labels and actual information. Inputs should be to the right and the labels on the left – right-aligned though so they are better associated with the input.
- [ ] opening and closing a contact should have a quick expand and compress animation so you see better what’s going on
- [ ] when multiple contacts are selected, currently you need to scroll up to get to the actions. Instead it should be like in Files – as soon as multiple contacts are selected, the top bar should appear fixed on the top.
- [ ] for the »Groups«, please use the multiselect.js dropdown which we also use for the groups in the ownCloud user management.
- [ ] when adding a new contact, there is both a »Cancel« button on the bottom and a »Delete x« on the top right. A »Cancel x« on the top right would suffice, no need for a button on the bottom.
- [ ] the default fields for a new contact (or one you edit) should be better: Name, Organization, Phone number, Email, Address (see #238)
- [ ] it takes some time to switch from a smaller group (with ~5 contacts) to a bigger group (with ~200 contacts). For that either it needs to be sped up, or there needs to be a progress spinner displayed in the main area.

That’s it for now, more coming in the next days while I use it more. @jbtbnl hope that’s helpful! :)

**Comment by jbtbnl** [comment]
@jancborchardt @tanghus I believe we should discuss what issues are to be part of this pull request and what issues are to be treated individually. Mainly because I am starting to lose oversight :)

In my opinion remarks 4,6,7,8,9 of @jancborchardt qualify to be issues to owncloud/master.

**Comment by jbtbnl** [comment]
@jancborchardt great remarks by the way, and thanks for running this branch :+1:

**Comment by jancborchardt** [comment]
@jbtbnl good point. Maybe we should actually merge this branch and then work on the individual issues @tanghus? Otherwise they might diverge too far. It’s a great improvement over the current interface.

**Comment by jbtbnl** [comment]
@tanghus https://github.com/owncloud/contacts/pull/290#issuecomment-30878189 => **It seems the picture loading for open contacts is broken? I see no pictures for any contacts.**

I noticed this bug also occurs in owncloud/master, can you confirm this?

**Comment by jancborchardt** [comment]
@jbtbnl I added checkboxes to my previous comment so you can check off what you fixed. :) If you have time, it would be cool if you can open new separate issues for the points you consider issues in master – and then check them off here.

Cheers! :)

**Comment by jbtbnl** [comment]
@jancborchardt it seems I am not able to check them off...

**Comment by jancborchardt** [comment]
Hm, maybe because you are not added as contributor to this repo. @karlitschek can you add @jbtbnl as contributor to the contacts repo?

**Comment by jancborchardt** [comment]
@jbtbnl can you rebase your branch? It seems to have been diverged from master – sorry about that.

@tanghus @jbtbnl can we merge this pull request after that? Or are there any breaking issues not present in master?

**Comment by jancborchardt** [comment]
@jbtbnl while you can’t check these checkboxes off, can you maybe collect those points which are left and either (when it’s a master issue) open a dedicated issue for it in the tracker or (when it’s an issue with your branch here) list them in a comment here so we can check them out.

Thanks man! :)

**Comment by jbtbnl** [comment]
@jancborchardt 

> the number of contacts in a group is in () parentheses. No idea why, and these can be removed.

Already fixed.

> the input on the bottom right of the user image is cool. But the whole user image should be clickable to select the contact. You can solve this like it’s done in Files: With a label which has enough size/padding to cover that area.

Fixed, see also #326.

> the inputs and labels should really not be in the same column – this creates a mix of labels and actual information. Inputs should be to the right and the labels on the left – right-aligned though so they are better associated with the input.

Better create a separate issue for this as we need to discuss the exact location of not only the labels but also the property type selector and hover actions. I guess it should be reconsidered using hover actions at all because it is unfriendly for touchscreens.

> opening and closing a contact should have a quick expand and compress animation so you see better what’s going on

Refer to #341.

> when multiple contacts are selected, currently you need to scroll up to get to the actions. Instead it should be like in Files – as soon as multiple contacts are selected, the top bar should appear fixed on the top.

Refer to #342.

> for the »Groups«, please use the multiselect.js dropdown which we also use for the groups in the ownCloud user management.

Refer to #343.

> when adding a new contact, there is both a »Cancel« button on the bottom and a »Delete x« on the top right. A »Cancel x« on the top right would suffice, no need for a button on the bottom.

Fixed.

> the default fields for a new contact (or one you edit) should be better: Name, Organization, Phone number, Email, Address (see #238)

We can continue this at issue #238.

> it takes some time to switch from a smaller group (with ~5 contacts) to a bigger group (with ~200 contacts). For that either it needs to be sped up, or there needs to be a progress spinner displayed in the main area.

I believe paginating contacts is the way to go, let's continue at https://github.com/owncloud/contacts/pull/337.

**Comment by MorrisJobke** [comment]
@jbtbnl I edited your comment and changed the bolded text to quoted text for a better understanding and distinction. Can I ask you to open the separate issues? (and add the issue number to the above comment ;))

**Comment by jancborchardt** [comment]
Ok, from my side we can merge this branch and work on the remaining issues separately. @tanghus @babelouest?

@owncloud/designers if you like, a review of this Contacts redesign would be cool as well. :)

**Comment by MorrisJobke** [comment]
Looks really good :+1: from my side

**Comment by MorrisJobke** [comment]
The header which appears when you select one or more contact should be fixed to the top and don't scroll away, but this could be fixed afterwards

**Comment by jbtbnl** [comment]
To summarize, after merging this branch issues #241 #278 #287 #327 #326 #271 #332 can be closed.

**Comment by tanghus** [comment]
> To summarize, after merging this branch issues #241 #278 #287 #327 #326 #271 #332 can be closed.

Which is awesome! I'll do a quick run-through tomorrow, but looks good so far.

@jbtbnl Can you make it so your commits don't change between 'Jan ten Bokkel' and @jbtbnl ? It makes git history very confusing.

**Comment by jancborchardt** [comment]
@tanghus any objections against merging this in? Otherwise I would do that later today. :)

**Comment by tanghus** [comment]
It looks very good - you've got my blessing :smile:

**Comment by jancborchardt** [comment]
Yeeeeehaaa, and the changes are in @jbtbnl :) huge thanks!

@jbtbnl btw if you use IRC, you should join our #owncloud-design channel on freenode. :) And if you want to dive more into ownCloud design, there’s http://owncloud.org/developer/design/ to get you started.

**Comment by jbtbnl** [comment]
Nice :)

**Pull request merged**